### PR TITLE
add notifications to mtv workflows

### PR DESCRIPTION
**Pull request by ydayagi**
FLPATH-1399
https://issues.redhat.com/browse/FLPATH-1399

### CODE REVIEW

File: mtv-migration/mtv.sw.yaml [hunk 1]
```diff
@@ -69,12 +75,36 @@ states:
...
+      - name: createNotification
+        functionRef:
+          refName: createNotification
+          arguments:
+            recipients:
+              type: "entity"
+              entityRef: .recipients
+            payload:
+              title: "MTV migration successful"
+              description: '"MTV migration " + .migrationName + " succeeded"'
```
Comment (pkliczewski) [inline mtv-migration/mtv.sw.yaml:105]: how difficult would it be to add a link to a virtual machine?

Comment (masayag) [reply mtv-migration/mtv.sw.yaml:105]: we actually did something like that for create-vm workflow:
https://github.com/parodos-dev/serverless-workflows/blob/main/request-vm-cnv/request-vm-cnv.sw.yaml#L243

Comment (ydayagi) [reply mtv-migration/mtv.sw.yaml:105]: it is a list of VMs. not a single VM.

Comment (pkliczewski) [reply mtv-migration/mtv.sw.yaml:105]: is there other way to reference the outcome of the migration?

Comment (ydayagi) [reply mtv-migration/mtv.sw.yaml:105]: not that I know of. I can try adding a link to the plan. not certain I have access to the OCP URL

File: mtv-migration/application.properties [hunk 2]
```diff
@@ -1,3 +1,5 @@
+quarkus.rest-client.notifications.url=${BACKSTAGE_NOTIFICATIONS_URL:http://backstage-backstage.rhdh-operator/api/notifications}
```
Comment (gabriel-farache) [inline mtv-migration/application.properties:1]: A detail but still: with https://github.com/parodos-dev/serverless-workflows/pull/307 there is a path in the openapi spec, the URL shall not contain the path; the default value should be `http://backstage-backstage.rhdh-operator`

File: mtv-plan/application.properties [hunk 3]
```diff
@@ -1,3 +1,5 @@
+quarkus.rest-client.notifications.url=${BACKSTAGE_NOTIFICATIONS_URL:http://backstage-backstage.rhdh-operator/api/notifications}
```
Comment (gabriel-farache) [inline mtv-plan/application.properties:1]: same: with https://github.com/parodos-dev/serverless-workflows/pull/307 there is a path in the openapi spec, the URL shall not contain the path; the default value should be http://backstage-backstage.rhdh-operator

File: mtv-plan/mtv.sw.yaml [hunk 4]
```diff
@@ -82,12 +88,36 @@ states:
...
+          refName: createNotification
+          arguments:
+            recipients:
+              type: "entity"
+              entityRef: .recipients
+            payload:
+              title: "MTV plan is ready for migration"
+              description: '"MTV plan " + .migrationName + " is ready for migration"'
+              scope: "SonataFlow"
+              topic: "MTV plan assessment workflow"
```
Comment (gabriel-farache) [inline mtv-plan/mtv.sw.yaml:120]: not aware of the details of the plan output, but is the plan available somewhere? If so, we might want to add a link to it (same as we do for MTA analysis output)

**Review by gabriel-farache — COMMENTED** [review]
some minor comments

File: mtv-plan/mtv.sw.yaml (hunk 4 above)
Comment (ydayagi) [reply mtv-plan/mtv.sw.yaml:120]: see the same comment by @pkliczewski i do not have a way to link it. i do not have the console URL.

File: mtv-migration/application.properties (hunk 2 above)
Comment (ydayagi) [reply mtv-migration/application.properties:1]: that's not minor. that's very important. thanks.

Comment (ydayagi) [reply mtv-migration/application.properties]: fixed

**Review by masayag — APPROVED** [review]

**Pull request merged**